src/main/java/io/apimatic/core/utilities/CoreHelper.java

```diff
@@ -338,16 +338,44 @@ public static String serializeTypeCombinator(Object obj) throws JsonProcessingEx
             return null;
         }
 
-        Annotation stringCaseAnnotation = obj.getClass()
-                .getAnnotation(TypeCombinatorStringCase.class);
-
-        if (stringCaseAnnotation != null) {
+        if (isTypeCombinatorStringCase(obj)) {
             return obj.toString();
         }
 
         return serialize(obj);
     }
 
+    /**
+     * Check either the object has TypeCombinatorStringCase annotation.
+     *
+     * @param obj  The object to check annotation.
+     * @return The boolean result.
+     */
+    public static boolean isTypeCombinatorStringCase(Object obj) {
```

Comment:
Question: Do we need to add unit tests for these newly added utilities?

Reply:
Added tests for these functions